### PR TITLE
Fix test 4.4

### PR DESCRIPTION
**Pull request by SchSeba**
* Update go mod to support ptp client
* Update sriov test to point sriov-tests
* Update ptp tests base on the ptp-operator project

### CODE REVIEW

**Comment by SchSeba** [comment]
ptp test output

```
export FEATURES_ENVIRONMENT=demo && export FEATURES="ptp" && make functests
Running Functional Tests
FEATURES="ptp" hack/run-functests.sh
~/Documents/GolangProjects/src/github.com/openshift-kni/cnf-features-deploy ~/Documents/GolangProjects/src/github.com/openshift-kni/cnf-features-deploy
/usr/bin/go
/home/sscheink/Documents/GolangProjects/bin/ginkgo
Running local tests
Focusing on ptp
Running Suite: CNF Features e2e integration tests
=================================================
Random Seed: 1588250073
Will run 9 of 18 specs

SSSSSSSSS•••
------------------------------
• [SLOW TEST:164.259 seconds]
[ptp]
/home/sscheink/Documents/GolangProjects/src/github.com/openshift-kni/cnf-features-deploy/functests/ptp/ptp.go:56
  PTP e2e tests
  /home/sscheink/Documents/GolangProjects/src/github.com/openshift-kni/cnf-features-deploy/functests/ptp/ptp.go:112
    PTP Interfaces discovery
    /home/sscheink/Documents/GolangProjects/src/github.com/openshift-kni/cnf-features-deploy/functests/ptp/ptp.go:196
      The interfaces support ptp can be discovered correctly
      /home/sscheink/Documents/GolangProjects/src/github.com/openshift-kni/cnf-features-deploy/functests/ptp/ptp.go:212
------------------------------
• [SLOW TEST:79.955 seconds]
[ptp]
/home/sscheink/Documents/GolangProjects/src/github.com/openshift-kni/cnf-features-deploy/functests/ptp/ptp.go:56
  PTP e2e tests
  /home/sscheink/Documents/GolangProjects/src/github.com/openshift-kni/cnf-features-deploy/functests/ptp/ptp.go:112
    PTP Interfaces discovery
    /home/sscheink/Documents/GolangProjects/src/github.com/openshift-kni/cnf-features-deploy/functests/ptp/ptp.go:196
      The virtual interfaces should be not discovered by ptp
      /home/sscheink/Documents/GolangProjects/src/github.com/openshift-kni/cnf-features-deploy/functests/ptp/ptp.go:224
------------------------------
••
------------------------------
• Failure [4.933 seconds]
[ptp]
/home/sscheink/Documents/GolangProjects/src/github.com/openshift-kni/cnf-features-deploy/functests/ptp/ptp.go:56
  PTP e2e tests
  /home/sscheink/Documents/GolangProjects/src/github.com/openshift-kni/cnf-features-deploy/functests/ptp/ptp.go:112
    PTP Interfaces discovery
    /home/sscheink/Documents/GolangProjects/src/github.com/openshift-kni/cnf-features-deploy/functests/ptp/ptp.go:196
      PTP time diff between Grandmaster and Slave should be in range -100ms and 100ms [It]
      /home/sscheink/Documents/GolangProjects/src/github.com/openshift-kni/cnf-features-deploy/functests/ptp/ptp.go:284

      Expected
          <int>: 239
      to be <
          <int>: 100

      /home/sscheink/Documents/GolangProjects/src/github.com/openshift-kni/cnf-features-deploy/functests/ptp/ptp.go:373
------------------------------
• [SLOW TEST:94.979 seconds]
[ptp]
/home/sscheink/Documents/GolangProjects/src/github.com/openshift-kni/cnf-features-deploy/functests/ptp/ptp.go:56
  Test NodeName Selector And Policy Priority
  /home/sscheink/Documents/GolangProjects/src/github.com/openshift-kni/cnf-features-deploy/functests/ptp/ptp.go:304
    Check if Node has Profile
    /home/sscheink/Documents/GolangProjects/src/github.com/openshift-kni/cnf-features-deploy/functests/ptp/ptp.go:335
      PTP daemon can apply match rule based on nodeLabel
      /home/sscheink/Documents/GolangProjects/src/github.com/openshift-kni/cnf-features-deploy/functests/ptp/ptp.go:337
------------------------------

JUnit report was created: /tmp/artifacts/unit_report_local.xml


Summarizing 1 Failure:

[Fail] [ptp] PTP e2e tests PTP Interfaces discovery [It] PTP time diff between Grandmaster and Slave should be in range -100ms and 100ms 
/home/sscheink/Documents/GolangProjects/src/github.com/openshift-kni/cnf-features-deploy/functests/ptp/ptp.go:373

Ran 9 of 18 Specs in 361.806 seconds
FAIL! -- 8 Passed | 1 Failed | 0 Pending | 9 Skipped
--- FAIL: TestTest (361.81s)
FAIL

Ginkgo ran 1 suite in 6m11.754697937s
Test Suite Failed
Running external tests
[INFO] Feature 'ptp' does not have external tests entry point
[WARN] Tests failed:
Tier 2 tests for ptp
make: *** [functests] Error 1

```

**Comment by SchSeba** [comment]
```
Running Functional Tests
FEATURES="sriov" hack/run-functests.sh
~/Documents/GolangProjects/src/github.com/openshift-kni/cnf-features-deploy ~/Documents/GolangProjects/src/github.com/openshift-kni/cnf-features-deploy
/usr/bin/go
/home/sscheink/Documents/GolangProjects/bin/ginkgo
Running local tests
Focusing on sriov
Running Suite: CNF Features e2e integration tests
=================================================
Random Seed: 1588254229
Will run 0 of 18 specs

SSSSSSSSSSSSSSSSSS
JUnit report was created: /tmp/artifacts/unit_report_local.xml

Ran 0 of 18 Specs in 6.733 seconds
SUCCESS! -- 0 Passed | 0 Failed | 0 Pending | 18 Skipped
PASS

Ginkgo ran 1 suite in 14.757334827s
Test Suite Passed
Running external tests
[INFO] Running external tests for sriov
Using latest master branch commit for https://github.com/openshift/sriov-tests
https://github.com/openshift/sriov-tests commit hash: 189e0fb45a7b8224e3c8e6bc78198cbd89b0eda9
Cloning code from https://github.com/openshift/sriov-tests using hash 189e0fb45a7b8224e3c8e6bc78198cbd89b0eda9
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   163  100   163    0     0    156      0  0:00:01  0:00:01 --:--:--   156
  0     0    0 6578k    0     0  1398k      0 --:--:--  0:00:04 --:--:-- 1860k
make[1]: Entering directory `/tmp/sriov-tests'
./scripts/run-conformance.sh
/home/sscheink/Documents/GolangProjects/bin/ginkgo
Running Suite: SRIOV Operator conformance tests
===============================================
Random Seed: 1588254250
Will run 13 of 13 specs

• [SLOW TEST:147.302 seconds]
pod
/tmp/sriov-tests/conformance/tests/vfs_assignement.go:25
  Configuration
  /tmp/sriov-tests/conformance/tests/vfs_assignement.go:39
    Virtual Functions
    /tmp/sriov-tests/conformance/tests/vfs_assignement.go:86
      should release the VFs once the pod deleted and same VFs can be used by the new created pods
      /tmp/sriov-tests/conformance/tests/vfs_assignement.go:88
------------------------------
• [SLOW TEST:76.801 seconds]
operator
/tmp/sriov-tests/conformance/tests/sriov_operator.go:31
  Configuration
  /tmp/sriov-tests/conformance/tests/sriov_operator.go:48
    SR-IOV network config daemon can be set by nodeselector
    /tmp/sriov-tests/conformance/tests/sriov_operator.go:50
      Should schedule the config daemon on selected nodes
      /tmp/sriov-tests/conformance/tests/sriov_operator.go:52
------------------------------
• [SLOW TEST:204.980 seconds]
operator
/tmp/sriov-tests/conformance/tests/sriov_operator.go:31
  Configuration
  /tmp/sriov-tests/conformance/tests/sriov_operator.go:48
    PF Partitioning
    /tmp/sriov-tests/conformance/tests/sriov_operator.go:106
      Should be possible to partition the pf's vfs
      /tmp/sriov-tests/conformance/tests/sriov_operator.go:108
------------------------------
S [SKIPPING] [93.811 seconds]
operator
/tmp/sriov-tests/conformance/tests/sriov_operator.go:31
  Configuration
  /tmp/sriov-tests/conformance/tests/sriov_operator.go:48
    PF Partitioning
    /tmp/sriov-tests/conformance/tests/sriov_operator.go:106
      Should not be possible to have overlapping pf ranges [It]
      /tmp/sriov-tests/conformance/tests/sriov_operator.go:224

      Overlapping is still not blocked

      /tmp/sriov-tests/conformance/tests/sriov_operator.go:227
------------------------------
• [SLOW TEST:153.959 seconds]
operator
/tmp/sriov-tests/conformance/tests/sriov_operator.go:31
  Configuration
  /tmp/sriov-tests/conformance/tests/sriov_operator.go:48
    VF flags
    /tmp/sriov-tests/conformance/tests/sriov_operator.go:292
      Should configure the spoofChk boolean variable
      /tmp/sriov-tests/conformance/tests/sriov_operator.go:420
------------------------------
• [SLOW TEST:200.593 seconds]
operator
/tmp/sriov-tests/conformance/tests/sriov_operator.go:31
  Configuration
  /tmp/sriov-tests/conformance/tests/sriov_operator.go:48
    VF flags
    /tmp/sriov-tests/conformance/tests/sriov_operator.go:292
      Should configure the trust boolean variable
      /tmp/sriov-tests/conformance/tests/sriov_operator.go:465
------------------------------
• [SLOW TEST:240.414 seconds]
operator
/tmp/sriov-tests/conformance/tests/sriov_operator.go:31
  Configuration
  /tmp/sriov-tests/conformance/tests/sriov_operator.go:48
    VF flags
    /tmp/sriov-tests/conformance/tests/sriov_operator.go:292
      Should configure the the link state variable
      /tmp/sriov-tests/conformance/tests/sriov_operator.go:509
------------------------------
• [SLOW TEST:208.838 seconds]
operator
/tmp/sriov-tests/conformance/tests/sriov_operator.go:31
  Configuration
  /tmp/sriov-tests/conformance/tests/sriov_operator.go:48
    VF flags
    /tmp/sriov-tests/conformance/tests/sriov_operator.go:292
      rate limit
      /tmp/sriov-tests/conformance/tests/sriov_operator.go:572
        Should configure the requested rate limit flags under the vf
        /tmp/sriov-tests/conformance/tests/sriov_operator.go:573
------------------------------
• [SLOW TEST:190.292 seconds]
operator
/tmp/sriov-tests/conformance/tests/sriov_operator.go:31
  Configuration
  /tmp/sriov-tests/conformance/tests/sriov_operator.go:48
    VF flags
    /tmp/sriov-tests/conformance/tests/sriov_operator.go:292
      vlan and Qos vlan
      /tmp/sriov-tests/conformance/tests/sriov_operator.go:618
        Should configure the requested vlan and Qos vlan flags under the vf
        /tmp/sriov-tests/conformance/tests/sriov_operator.go:619
------------------------------
• [SLOW TEST:181.303 seconds]
operator
/tmp/sriov-tests/conformance/tests/sriov_operator.go:31
  Configuration
  /tmp/sriov-tests/conformance/tests/sriov_operator.go:48
    Resource Injector
    /tmp/sriov-tests/conformance/tests/sriov_operator.go:653
      Should inject downward api volume
      /tmp/sriov-tests/conformance/tests/sriov_operator.go:655
------------------------------
• [SLOW TEST:182.129 seconds]
operator
/tmp/sriov-tests/conformance/tests/sriov_operator.go:31
  Configuration
  /tmp/sriov-tests/conformance/tests/sriov_operator.go:48
    Resource Injector
    /tmp/sriov-tests/conformance/tests/sriov_operator.go:653
      Multiple sriov device and attachment
      /tmp/sriov-tests/conformance/tests/sriov_operator.go:747
        Should configure multiple network attachments
        /tmp/sriov-tests/conformance/tests/sriov_operator.go:749
------------------------------
• [SLOW TEST:199.056 seconds]
operator
/tmp/sriov-tests/conformance/tests/sriov_operator.go:31
  Configuration
  /tmp/sriov-tests/conformance/tests/sriov_operator.go:48
    Resource Injector
    /tmp/sriov-tests/conformance/tests/sriov_operator.go:653
      IPv6 configured secondary interfaces on pods
      /tmp/sriov-tests/conformance/tests/sriov_operator.go:774
        should be able to ping each other
        /tmp/sriov-tests/conformance/tests/sriov_operator.go:776
------------------------------
• [SLOW TEST:199.178 seconds]
operator
/tmp/sriov-tests/conformance/tests/sriov_operator.go:31
  Configuration
  /tmp/sriov-tests/conformance/tests/sriov_operator.go:48
    MTU
    /tmp/sriov-tests/conformance/tests/sriov_operator.go:803
      Should support jumbo frames
      /tmp/sriov-tests/conformance/tests/sriov_operator.go:864
------------------------------

JUnit report was created: /tmp/artifacts/unit_report.xml

Ran 12 of 13 Specs in 2278.665 seconds
SUCCESS! -- 12 Passed | 0 Failed | 0 Pending | 1 Skipped
PASS

Ginkgo ran 1 suite in 38m9.450736101s
Test Suite Passed
make[1]: Leaving directory `/tmp/sriov-tests'
~/Documents/GolangProjects/src/github.com/openshift-kni/cnf-features-deploy

```

File: functests/temp/ptp-tmp.go [hunk 1]
```diff
@@ -0,0 +1,492 @@
...
+			}
+			return nil
+		}, 2*time.Minute, 1*time.Second).Should(BeNil())
+		if isPTPEnabled(&buf) == false {
+			ptpSupportedInterfaces = append(ptpSupportedInterfaces, interf)
+		}
+		time.Sleep(time.Second)
+	}
+	return ptpSupportedInterfaces
+}
```
Comment (yuvalk) [inline functests/temp/ptp-tmp.go:492]: why is this temp?

Comment (SchSeba) [reply functests/temp/ptp-tmp.go:492]: I miss that here is a PR to fix it

https://github.com/openshift-kni/cnf-features-deploy/pull/174

Thanks for finding it :)

Comment (yuvalk) [reply functests/temp/ptp-tmp.go:492]: try to avoid 
git add -a 
next time ;-)

**Pull request merged**